src/components/portfolio/DetailsMainAnalyticsCorrelations.tsx

```diff
@@ -0,0 +1,156 @@
+
+  // gets unique set of companies displayed in the chart (used as x-axis categories)
+  const chartCategories = Array.from(
+    new Set(
+      mappedCorrelations
+        .map((mc) => [mc[0], mc[1]])
+        .reduce((arr1, arr2) => arr1.concat(arr2))
+        .sort()
+    )
+  );
+
+  // will be passed to the chart as series prop
+  const finalSeries: HeatMapSeries[] = [];
+
+  chartCategories.forEach((c1) => {
+    const currentSeries: HeatMapSeries = {
+      name: c1,
+      data: [],
+    };
+    chartCategories.forEach((c2) => {
+      // the correlation of a company to itself is 1
+      if (c1 === c2) currentSeries.data.push(1);
+      else {
+        /* check if c1 + c2 are present as a pair in the mappedCorrelations ignoring order
+         if present push the corresponding correlation value, if no match is found place default value 0
+         it is mandatory that each series has the same amount and order of ordered values to display a meaningful x-Axis in the chart */
+        let valueToPush = 0;
+        mappedCorrelations.forEach((mc) => {
+          if (
+            JSON.stringify([c1, c2].sort()) ===
+            JSON.stringify(mc.slice(0, 2).sort())
+          )
+            // eslint-disable-next-line prefer-destructuring
+            valueToPush = mc[2];
+        });
+        currentSeries.data.push(valueToPush);
+      }
+    });
+    finalSeries.push(currentSeries);
+  });
+
+  const options = {
+    chart: {
+      type: 'heatmap',
+      toolbar: {
+        show: false,
+      },
+    },
+    dataLabels: {
+      enabled: false,
+    },
+    colors: [theme.palette.lightBlue.main],
+    title: {
+      text: t('portfolio.details.analytics.correlations.chartTitle'),
+      style: {
+        color: theme.palette.primary.contrastText,
+      },
+    },
+    xaxis: {
+      categories: chartCategories,
+      labels: {
+        style: {
+          colors: theme.palette.primary.contrastText,
+        },
+      },
+    },
+    yaxis: {
+      labels: {
+        style: {
+          colors: theme.palette.primary.contrastText,
+        },
+      },
+    },
+  };
+
+  return (
+    <Chart
+      type="heatmap"
+      height={height}
+      series={finalSeries}
+      options={options}
+    />
+  );
+};
+
+type DetailsMainAnalyticsCorrelationsProps = {
+  portfolio: NonEmptyPortfolioDetails;
+};
+
+const DetailsMainAnalyticsCorrelations: React.FC<DetailsMainAnalyticsCorrelationsProps> = ({
+  portfolio,
+}) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+
+  return (
+    <div>
+      <div className={classes.standardDeviationWrapper}>
+        <span>{t('portfolio.details.analytics.standardDeviation')}</span>
+        <span className={classes.standardDeviationValue}>
+          <StyledNumberFormat
+            value={portfolio.analytics.standardDeviation * 100}
+            suffix="%"
+          />
+        </span>
+      </div>
+      <Heatmap portfolio={portfolio} height={350} />
+    </div>
+  );
+};
```

Comment:
Why not pull this up a level into the otherwise empty DetailsMainAnalytics? Then this file could just be the heatmap.

Reply:
Good idea, done.

Reply:
Nice